### ISSUE TITLE
Wrong check in kernel name validation

### ISSUE THREAD

**Issue by pith** [nuun-io/kernel#44]
The `KernelCore` constructor [checks](https://github.com/nuun-io/kernel/blob/e9388d38cfdbcf3c1f20c18aa7de9ddd0d554ec8/core/src/main/java/io/nuun/kernel/core/internal/KernelCore.java#L109) that the kernel name is unique. The actual check doesn't work, it should be replaced by `kernels.containsKey(name())`.

```
// private static final ConcurrentHashMap<String, Kernel> kernels  = new ConcurrentHashMap<String, Kernel>();

if (!kernels.contains(name()))
{
    kernels.put(name(), this);
}
else
{
    throw new KernelException(String.format("A kernel named %s already exists", name()));
}
```

Otherwise, it seems that this map is only used to check the name uniqueness. So is it really necessary to store kernel instances in the map ?

Currently, the name cannot be configured. So maybe, if we could have a safer name initialization, the map could be removed. For instance at this [line](https://github.com/nuun-io/kernel/blob/e9388d38cfdbcf3c1f20c18aa7de9ddd0d554ec8/core/src/main/java/io/nuun/kernel/core/internal/KernelCore.java#L105), we could use an `AtomicInteger`.

    name = KERNEL_PREFIX_NAME + kernelIndex.incrementAndGet();

**Comment by ejemba** [comment]
This work has been initiated for the multi-kernel purpose.
As for now research on the subject is always ongoing, you're right we could get ride of it for now.

**Comment by pith** [comment]
Will be fixed in 72e8919297d8b0be57f989e63302c1af780ac767
